### DISCUSSION TITLE
Use the ACR Module

### DISCUSSION THREAD

**Discussion by pualfr** [jrkerns/pylinac#463] [Q&A]
Hi everyone,

I tried to test the ACRMRILarge module.
I have put 11 MRI slices of a dedicated acquisition in a folder and I try to pull these lines:

from pylinac import ACRMRILarge
acr_folder = r"C:/Temp/ACR"
mri = ACRMRILarge(acr_folder)
mri.results()

and when a enter the mri.result() I have the following problem:

> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "C:\Users\mz-u2259\AppData\Local\Programs\Python\Python311\Lib\site-packages\pylinac\acr.py", line 1134, in results
>     # plot the side view
> 
> AttributeError: 'ACRMRILarge' object has no attribute 'geometric_distortion'

Is there any specific acquisition protocol or code to make is work?
Thanks a lot,
Paul

**Comment by jrkerns** [comment]
As described in the typical use section (https://pylinac.readthedocs.io/en/latest/acr.html#typical-use), you need to analyze the phantom first.  So:

`mri = ACRMRILarge(...)`
`mri.analyze()`
`print(mri.results())`

**Reply by pualfr** [reply]
Actually I tried the analyze() first and get an even longer error message, I'll paste it here tomorrow when I have access to my test setup.

**Reply by pualfr** [reply]
Hi again and thanks for the initial reply!
When running the analyze function I have these messages:

> >>> from pylinac import ACRMRILarge
>>> acr_folder = r"C:/Temp/ACR"
>>> mri = ACRMRILarge(acr_folder)
>>> mri.analyze()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\mz-u2259\AppData\Local\Programs\Python\Python311\Lib\site-packages\pylinac\acr.py", line 1034, in analyze
    self.localize()
  File "C:\Users\mz-u2259\AppData\Local\Programs\Python\Python311\Lib\site-packages\pylinac\acr.py", line 974, in localize
    raise ValueError(
ValueError: The physical scan extent does not cover the extent of module configuration. This means not all modules were included in the scan. Rescan the phantom to include all relevant modules, or change the offset values.

My initial image set has 11 images corresponding to the ACR scanning protocol.
Also I checked for the 3 offsets and they seem to be correct if I assume that the first slice offset is 0 mm.

Thanks a lot!
Paul

**Reply by jrkerns** [reply]
You can submit them here and I'll take a look: https://forms.gle/CZGJBd1PjPk4xAsc9

**Reply by pualfr** [reply]
Thanks a lot I just uploaded the images.

Paul

**Comment by jrkerns** [comment]
There are two bugs I can see. The configuration extent check has a rounding error. Note the 53.2738 vs 53.2739. This is obviously rounding differences. 

![Screenshot 2023-08-30 150916](https://github.com/jrkerns/pylinac/assets/4803336/9bb6ed88-d613-4224-9d80-2a320af277ec)

This can be worked around using this snippet: https://gist.github.com/jrkerns/073c72d6b55c20afc84a55029eb28410
This will let you **plot** the result:

```
acr = MyACR(...)
acr.analyze()
acr.plot_analyzed_image()
```

Which looks like so:

![Figure_1](https://github.com/jrkerns/pylinac/assets/4803336/d2664286-17a2-4ac1-8b30-02befaf470a1)

Unfortunately, the `.results()` call still fails. This is due to a more subtle bug lower down in the code. I will have a fix for both out in the next release, due 9/21 if not before then. Thanks for reporting. 

It also looks like the geometric module is a bit different than what I can see from the ACR docs (Figure 4, page 8 of https://www.acraccreditation.org/-/media/ACRAccreditation/Documents/MRI/LargePhantomGuidance.pdf?la=en). What version of the phantom is this?

**Comment by pualfr** [comment]
Thank you very much for the feedback.
Regarding the phantom, I cannot find the model on it but the geometric module is definitely different. Visually, the phantom looks like this one: https://www.newmaticmedical.com/product/ACR-MRI-Phantom-ACRMRP

**Comment by aljosapolsak** [comment]
Hi
I am also testing ACR module, but I use it integrated to QATrack+. We are currently using pylinac 3.30. I get images that and distanes that seem off -geomteric distortion. 
<img width="2233" height="1242" alt="mri" src="https://github.com/user-attachments/assets/e39d8522-4221-4e48-be29-5edbaac4bed8" />

May I ask for suggestion? Thank you
